nncf/data/utils.py

```diff
@@ -0,0 +1,74 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import List
+from typing import Iterator
+from typing import Callable
+
+from nncf.data.types import DataSource
+from nncf.data.types import DataItem
+from nncf.data.types import ModelInput
+from nncf.data.dataloader import DataLoader
+from nncf.data.dataloader import DataLoaderImpl
+
+
+def create_dataloader(data_source: DataSource,
+                      transform_fn: Callable[[DataItem], ModelInput]) -> DataLoader:
+    """
```

Comment:
Is it possible to make transform function optional? In some cases dataloader can be defined specifically for quantization (e.g. for pipelined models, as example, we will not use dataloader for model part as loader for whole pipeline) and transformation logic can be part of __iter__ or __getitem__.

Reply:
Thanks for you comment. We will make it possible to pass ```None``` as ```transform_fn```

Reply:
It was supported.